pkg/controllers/dnspolicy/dnsrecord_eventhandler.go

```diff
@@ -0,0 +1,78 @@
+package dnspolicy
+
+import (
+	"context"
+
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/_internal/slice"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/apis/v1alpha1"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/controllers/gateway"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type DNSRecordEventHandler struct {
+	client      client.Client
+	hostService gateway.HostService
+}
+
+var _ handler.EventHandler = &DNSRecordEventHandler{}
+
+// Create implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.Object, q)
+}
+
+// Delete implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Delete(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.Object, q)
+}
+
+// Generic implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Generic(e event.GenericEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.Object, q)
+}
+
+// Update implements handler.EventHandler
+func (eh *DNSRecordEventHandler) Update(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	eh.enqueueForObject(e.ObjectNew, q)
+}
+
+func (eh *DNSRecordEventHandler) enqueueForObject(obj v1.Object, q workqueue.RateLimitingInterface) {
+	dnsRecord, ok := obj.(*v1alpha1.DNSRecord)
+	if !ok {
+		return
+	}
+
+	dnsPolicyList := &v1alpha1.DNSPolicyList{}
+	if err := eh.client.List(context.TODO(), dnsPolicyList); err != nil {
```

Comment:
is the intention to change this context eventually? If not, I think it's better to use `context.Background`

Reply:
functionally they are the same, but `context.Background` suggests the use was intentional where `context.TODO` suggests that it should be changed to something else eventually.